controllers/managedocs_controller.go

```diff
@@ -817,8 +817,12 @@ func (r *ManagedOCSReconciler) getDesiredProviderStorageCluster() (*ocsv1.Storag
 }
 
 func (r *ManagedOCSReconciler) getDesiredConsumerStorageCluster() (*ocsv1.StorageCluster, error) {
-	storageSize := r.addonParams[storageSizeKey]
-	storageUnit := r.addonParams[storageUnitKey]
+
+	// TODO: set max possible quota size
+	// re-visit if product enforces quota or any means respecting the storage claim
+	storageSize := fmt.Sprintf("%v", int32(math.MaxInt32))
+	storageUnit := "Ti"
```

Comment:
First, I think this is the wrong place to do the overwrite, I would prefer to override the addon param when we read it and not the local var we are using for calculations, mainly because we want to see/track the actual value we use in the various log message we print with addon params values.

Second, please limit the storage size to 1 or 2 Peta, any value bigger than that might cause issues in rook or ceph as they usually add a little bit on top (which is hard to do when you give the max value of the int type)

Taking both of the above points into consideration, I would suggest to implement the following change here https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/controllers/managedocs_controller.go#L512

```
		for key, value := range addonParamSecret.Data {
			r.addonParams[key] = string(value)
		}
		
		// Deprecting consumer requested capacity addon parameter by overriding it 
		// with a very high pre-defined value
		if r.DeploymentType == consumerDeploymentType {
			r.Log.Info(fmt.Printf(
				`Overriding addon parameter "%s" (currently "%s") with 1024`,
				storageSizeKey,
				r.addonParams[storageSizeKey] 
			))
			r.addonParams[storageSizeKey] = "1024"

			r.Log.Info(fmt.Printf(
				`Overriding addon parameter "%s" (currently "%s") with Ti`,
				storageUnitKey,
				r.addonParams[storageUnitKey] 
			))
			r.addonParams[storageUnitKey] = "Ti"
		}
```

Reply:
Looks neat and thanks for explanations.